File: pipelines_webinar/webservice_stage.py
from aws_cdk import core

from .pipelines_webinar_stack import PipelinesWebinarStack

class WebServiceStage(core.Stage):
  def __init__(self, scope: core.Construct, id: str, **kwargs):
    super().__init__(scope, id, **kwargs)

    service = PipelinesWebinarStack(self, 'WebService')

    self.url_output = service.url_output

File: pipelines_webinar/webservice_stage.ts
import { Construct } from 'constructs';
import { CfnOutput, StackProps, Stage } from 'aws-cdk-lib';
import { PipelinesWebinarStack } from './pipelines_webinar_stack';

export enum ENV {
  SDLC,
  PROD
}

export interface WebServiceStageProps {
  environment: ENV
}

export class WebServiceStage extends Stage {
  urlOutput: CfnOutput;

  constructor(scope: Construct, id: string, props: WebServiceStageProps) {
    super(scope, id);

    const service = new PipelinesWebinarStack(this, 'WebService', {
      tags: {
        Application: 'WebService',
        Environment: id
      },
      environment: props.environment
    });

    this.urlOutput = service.urlOutput
  }
}
